### DISCUSSION TITLE
Force object to get focus

### DISCUSSION THREAD

**Discussion by ABuffEr** [nvaccess/nvda#13122] [Q&A]
Hi all, 

I asked on nvda-addons, but the solution may be too advanced/time consuming to find from scratch, and I received no anwser. I'm particularly interested to an answer from @feerrenrut, @michaelDCurran, @leonardder or maybe @seanbudd too, who worked heavily on UIA.

I have strange problem focusing an object. If you have Word (2019, but with older or newer could be the same, I don't know), and you create this code in scratchpad: 

```
import globalPluginHandler
import api
import winUser
import mouseHandler
from scriptHandler import script

class GlobalPlugin(globalPluginHandler.GlobalPlugin):

    @script(
        gesture="kb:NVDA+i" # or what you want
    )
    def script_objInfo(self, gesture):
        obj = api.getNavigatorObject()
        winUser.setFocus(obj.windowHandle)
        api.setFocusObject(obj)
        (left, top, width, height) = obj.location
        x = left+(width//2)
        y = top+(height//2)
        winUser.setCursorPos(x, y)
        mouseHandler.executeMouseMoveEvent(x, y)
        obj.scrollIntoView()
        obj.setFocus() # or obj.UIAElement.setFocus()
```

then: 

1) open Word, open menu, tab Home, expand it; 
2) using object navigation, find Edit group (sorry if it's called differently, I have it in Italian); 
3) always with object navigation, locate the child Edit, not expanded/collapsed; 
4) execute the above script, then press NVDA+F1. 

As you'll see, isFocusable is True, but, regardless various attempts (including obj.setFocus() or obj.UIAElement.setFocus() that generate a COMError), hasFocus remains False. 

Obviously, this not happens (focus gets correctly) if you use the menu in the standard way.

Sorry for tags, indeed, but I'm writing an add-on and this situation caused me a lot of problems. I'll very grateful if you have time and suggestions about how to fix it.

**Comment by bdorer** [comment]
This is a question for the nvda-devel mailinglist I think.

**Reply by ABuffEr** [reply]
> This is a question for the nvda-devel mailinglist I think.

Yes, but I doubt it would have received any answer.
Anyway, I solved getting focus on a previous focusable object, and then simulating a "tab" keystroke, but it's not a so secure solution, imho.
